src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,176 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = newNode;
```

Comment:
with such implementation your head will have tail as prev, which may cause unpredicted behaviour in some cases